streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBRangeIteratorTest.java

```diff
@@ -0,0 +1,447 @@
+            key3Bytes,
+            toBytes,
+            false,
+            true
+        );
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.peekNextKey(), is(key4Bytes));
+        assertThat(rocksDBRangeIterator.peekNextKey(), is(key4Bytes));
+        assertThat(rocksDBRangeIterator.next().key, is(key4Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.peekNextKey(), is(key3Bytes));
+        assertThat(rocksDBRangeIterator.peekNextKey(), is(key3Bytes));
+        assertThat(rocksDBRangeIterator.next().key, is(key3Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(false));
+        assertThrows(NoSuchElementException.class, rocksDBRangeIterator::peekNextKey);
+        verify(rocksIterator);
+    }
+
+    @Test
+    public void shouldCloseIterator() {
+        final RocksIterator rocksIterator = mock(RocksIterator.class);
+        rocksIterator.seek(key1Bytes.get());
+        rocksIterator.close();
+        expectLastCall().times(1);
+        replay(rocksIterator);
+        final RocksDBRangeIterator rocksDBRangeIterator = new RocksDBRangeIterator(
+            storeName,
+            rocksIterator,
+            Collections.emptySet(),
+            key1Bytes,
+            key2Bytes,
+            true,
+            true
+        );
+        rocksDBRangeIterator.close();
+        verify(rocksIterator);
+    }
+
+    @Test
+    public void shouldReturnNextKeyValueMatchingPrefix() {
+        final RocksIterator rocksIterator = mock(RocksIterator.class);
+        final Bytes prefixKey1 = Bytes.wrap("prefix-1".getBytes());
+        final Bytes prefixKey2 = Bytes.wrap("prefix-2".getBytes());
+        final Bytes prefix = Bytes.wrap("prefix".getBytes());
+        final Bytes toBytes = Bytes.increment(prefix);
+        rocksIterator.seek(prefix.get());
+        expect(rocksIterator.isValid())
+            .andReturn(true)
+            .andReturn(true)
+            .andReturn(false);
+        expect(rocksIterator.key())
+            .andReturn(prefixKey1.get())
+            .andReturn(prefixKey2.get());
+        expect(rocksIterator.value()).andReturn(valueBytes).times(2);
+        rocksIterator.next();
+        expectLastCall().times(2);
+        replay(rocksIterator);
+        final RocksDBRangeIterator rocksDBRangeIterator = new RocksDBRangeIterator(
+            storeName,
+            rocksIterator,
+            Collections.emptySet(),
+            prefix,
+            toBytes,
+            true,
+            false
+        );
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(prefixKey1));
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(prefixKey2));
+        assertThat(rocksDBRangeIterator.hasNext(), is(false));
+        verify(rocksIterator);
+    }
```

Comment:
If you change the `toInclusive` to `true` in this test, the test also passes. That means, that you are not testing `toInclusive` in this test.

I suggest the following test: 
```
    public void shouldExcludeEndOfRange() {
        final RocksIterator rocksIterator = mock(RocksIterator.class);
        rocksIterator.seek(key1Bytes.get());
        expect(rocksIterator.isValid())
            .andReturn(true)
            .andReturn(true);
        expect(rocksIterator.key())
            .andReturn(key1Bytes.get())
            .andReturn(key2Bytes.get());
        expect(rocksIterator.value()).andReturn(valueBytes).times(2);
        rocksIterator.next();
        expectLastCall().times(2);
        replay(rocksIterator);
        final RocksDBRangeIterator rocksDBRangeIterator = new RocksDBRangeIterator(
            storeName,
            rocksIterator,
            Collections.emptySet(),
            key1Bytes,
            key2Bytes,
            true,
            false
        );
        assertThat(rocksDBRangeIterator.hasNext(), is(true));
        assertThat(rocksDBRangeIterator.next().key, is(key1Bytes));
        assertThat(rocksDBRangeIterator.hasNext(), is(false));
        verify(rocksIterator);
    }
```

Reply:
@cadonna , added this test.